### PR TITLE
Update TM UI for multiple interfaces

### PR DESCRIPTION
**Pull request by shamrickus**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue. 

The Cache Stats tab of TM UI has been updated to reflect multiple interfaces. Each cache server can have any number of interfaces, as such each Cache Stat row now features a hidden table containing the information about it's interfaces. This can be seen (and hidden) by clicking on the parent row.

## Which Traffic Control components are affected by this PR?

- Traffic Monitor

## What is the best way to verify this PR?
Requires #4690 to work

Verify TM UI functions properly and that all interface information can be viewed for each server.  

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_monitor/static/script.js [hunk 1]
```diff
@@ -147,51 +147,136 @@ function getEvents() {
...
+
 	ajax("/api/cache-statuses", function(r) {
 		const servers = new Map(Object.entries(JSON.parse(r)));
+
+		// Sort by key (server name) so order doesn't change 'randomly'
+		servers[Symbol.iterator] = function* () {
+			yield *[...this.entries()].sort((serverCoupleA, serverCoupleB) => {
+				return serverCoupleA[0] < serverCoupleB[0];
+			});
+		};
```
[the remark this answers is not in the document]
Comment (shamrickus) [reply traffic_monitor/static/script.js:183]: You are correct, the object is resorted every time it is iterated.

File: traffic_monitor/static/script.js [hunk 2]
```diff
@@ -155,51 +155,129 @@ function getEvents() {
...
 */
 function getCacheStates() {
+	function parseIPAvailable(server, ipField) {
+		return server.status.indexOf("ONLINE") !== 0 ? server[ipField] : "N/A";
+	}
+
+	function parseBandwidth(server) {
+		if (Object.prototype.hasOwnProperty.call(server, "bandwidth_kbps")) {
+			const kbps = (server.bandwidth_kbps / kilobitsInMegabit).toFixed(2);
+			const max = numberStrWithCommas((server.bandwidth_capacity_kbps / kilobitsInMegabit).toFixed(0));
```
Comment (ocket8888) [inline traffic_monitor/static/script.js:165]: This doesn't check for `bandwidth_capacity_kbps` as an own property of `server`, so for interfaces with no KBPS thresholds - particularly astats servers - it outputs max as `NaN`.

File: traffic_monitor/static/script.js [hunk 3]
```diff
@@ -155,51 +155,129 @@ function getEvents() {
...
+			return `${kbps} / ${max}`;
+		} else {
+			return "N/A";
+		}
+	}
+	function parseStatusClass(server, row) {
+		if (Object.prototype.hasOwnProperty.call(server, "status")) {
+			if (server.status.indexOf("ADMIN_DOWN") !== -1 || server.status.indexOf("OFFLINE") !== -1) {
+				row.classList.add("warning");
+			} else if (!server.combined_available && server.status.indexOf("ONLINE") !== 0) {
```
Comment (ocket8888) [inline traffic_monitor/static/script.js:175]: Interfaces have `status`, but do not have `combined_available`, therefore `server.combined_available` is always `undefined` for interfaces. Their statuses are not Traffic Ops statuses, either. They report `available` when they are available, not `ONLINE` or `REPORTED`.

If I have a server with these interfaces:
```json
{
	"interfaces": {
		"eth0":{
			"status": "available",
			"status_poller": "stat",
			"bandwidth_kbps": 29,
			"available":true
		}
	}
}
```
its row will show in text that it is available, but will be highlighted red as though it weren't.

**Review by ocket8888 — APPROVED** [review]
Looks good, appears to work.

**Pull request merged**